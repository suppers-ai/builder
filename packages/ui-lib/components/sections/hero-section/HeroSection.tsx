import {
  ArrowRight,
  Code2,
  Download,
  Github,
  Layout,
  Palette,
  Palmtree,
  Star,
  Zap,
} from "lucide-preact";
import { Button } from "../../action/button/Button.tsx";

export function HeroSection() {
  return (
    <section class="relative bg-gradient-to-br from-primary/10 via-secondary/5 to-accent/10 overflow-hidden">
      {/* Video Background */}
      <video
        autoplay
        muted
        playsInline
        preload="metadata"
        class="absolute inset-0 w-full h-full object-cover z-0"
        onError={() => console.warn("Hero video failed to load")}
      >
        <source src="/hero.mp4" type="video/mp4" />
        Your browser does not support the video tag.
      </video>

      {/* Subtle overlay for text readability */}
      <div class="absolute inset-0 bg-black/20 z-10"></div>

      {/* Background decoration */}
      <div class="absolute inset-0 bg-grid-pattern opacity-5 z-20"></div>
      <div class="absolute top-0 right-0 w-96 h-96 bg-primary/20 rounded-full blur-3xl -translate-y-48 translate-x-48 z-20">
      </div>
      <div class="absolute bottom-0 left-0 w-96 h-96 bg-secondary/20 rounded-full blur-3xl translate-y-48 -translate-x-48 z-20">
      </div>

      <div class="relative px-4 lg:px-6 py-20 lg:py-32 z-30">
        <div class="max-w-6xl mx-auto text-center">
          {/* Badge */}
          <div class="inline-flex items-center gap-2 bg-black/60 backdrop-blur-sm border border-white/20 rounded-full px-4 py-2 text-sm font-medium mb-8 shadow-lg text-white">
            <div class="w-2 h-2 bg-green-400 rounded-full animate-pulse"></div>
            Built for Fresh 2.0
          </div>

          {/* Main heading */}
          <h1 class="text-4xl lg:text-6xl xl:text-7xl font-bold text-white mb-6 leading-tight drop-shadow-lg">
            Build Beautiful{" "}
            <span class="text-transparent bg-clip-text bg-gradient-to-r from-primary via-secondary to-accent">
              User Interfaces
            </span>{" "}
            with Fresh
          </h1>

          {/* CTA Buttons */}
          <div class="flex flex-col gap-4 justify-center mb-12">
            {/* Primary Actions */}
            <div class="flex flex-col sm:flex-row gap-4 justify-center">
              <Button
                as="a"
                href="/components"
                color="primary"
                size="md"
                class="group"
              >
                Browse Components
                <ArrowRight size={20} className="group-hover:translate-x-1 transition-transform" />
              </Button>
            </div>
          </div>

          {/* Social proof */}
          <div class="flex flex-wrap justify-center items-center gap-6 text-sm text-white/80">
            <div class="flex items-center gap-2 bg-white/10 backdrop-blur-sm rounded-full px-3 py-1">
              <Star size={16} className="text-yellow-400 fill-current" />
              <span>Trusted by developers</span>
            </div>
            <div class="flex items-center gap-2 bg-white/10 backdrop-blur-sm rounded-full px-3 py-1">
              <Zap size={16} className="text-blue-400" />
              <span>Lightning fast</span>
            </div>
            <div class="flex items-center gap-2 bg-white/10 backdrop-blur-sm rounded-full px-3 py-1">
              <Palette size={16} className="text-purple-400" />
              <span>Themeable</span>
            </div>
          </div>
        </div>
      </div>

      {/* Bottom wave */}
      <div class="absolute -bottom-px left-0 right-0 z-30">
        <svg
          viewBox="0 0 1440 120"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
          class="w-full h-auto block"
          preserveAspectRatio="none"
        >
          <path
            d="M0 120L60 105C120 90 240 60 360 45C480 30 600 30 720 37.5C840 45 960 60 1080 67.5C1200 75 1320 75 1380 75L1440 75V120H1380C1320 120 1200 120 1080 120C960 120 840 120 720 120C600 120 480 120 360 120C240 120 120 120 60 120H0Z"
            fill="currentColor"
            class="text-base-100"
          />
        </svg>
      </div>
    </section>
  );
}
