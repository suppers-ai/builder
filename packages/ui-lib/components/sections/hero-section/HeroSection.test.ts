import { assertEquals, assertStringIncludes } from "https://deno.land/std@0.224.0/assert/mod.ts";
import { assertSnapshot } from "https://deno.land/std@0.224.0/testing/snapshot.ts";
import { renderToString } from "preact-render-to-string";
import { HeroSection } from "./HeroSection.tsx";

Deno.test("HeroSection - basic rendering", () => {
  const html = renderToString(HeroSection({}));
  assertStringIncludes(html, "Professional");
  assertStringIncludes(html, "DaisyUI Components");
  assertStringIncludes(html, "section");
});

Deno.test("HeroSection - contains main heading", () => {
  const html = renderToString(HeroSection({}));

  // Check for main heading content
  assertStringIncludes(html, "Professional");
  assertStringIncludes(html, "DaisyUI Components");
  assertStringIncludes(html, "for Fresh");
});

Deno.test("HeroSection - contains subtitle description", () => {
  const html = renderToString(HeroSection({}));

  // Check for subtitle content
  assertStringIncludes(html, "The most comprehensive DaisyUI component library");
  assertStringIncludes(html, "Fresh 2.0");
  assertStringIncludes(html, "65+ production-ready components");
  assertStringIncludes(html, "TypeScript support");
});

Deno.test("HeroSection - contains stats section", () => {
  const html = renderToString(HeroSection({}));

  // Check for statistics
  assertStringIncludes(html, "65+");
  assertStringIncludes(html, "Components");
  assertStringIncludes(html, "100%");
  assertStringIncludes(html, "DaisyUI Coverage");
  assertStringIncludes(html, "29");
  assertStringIncludes(html, "Themes");
  assertStringIncludes(html, "TypeScript");
  assertStringIncludes(html, "Full Support");
});

Deno.test("HeroSection - contains CTA buttons", () => {
  const html = renderToString(HeroSection({}));

  // Check for call-to-action buttons
  assertStringIncludes(html, "Browse Components");
  assertStringIncludes(html, "Get Started");
  assertStringIncludes(html, "View on GitHub");
});

Deno.test("HeroSection - contains Fresh 2.0 badge", () => {
  const html = renderToString(HeroSection({}));

  // Check for Fresh 2.0 support badge
  assertStringIncludes(html, "New: Fresh 2.0 Support");
  assertStringIncludes(html, "animate-pulse");
});

Deno.test("HeroSection - contains social proof", () => {
  const html = renderToString(HeroSection({}));

  // Check for social proof elements
  assertStringIncludes(html, "Trusted by developers");
  assertStringIncludes(html, "Production ready");
  assertStringIncludes(html, "Lightning fast");
  assertStringIncludes(html, "Fully themeable");
});

Deno.test("HeroSection - has proper link structure", () => {
  const html = renderToString(HeroSection({}));

  // Check for proper link elements
  assertStringIncludes(html, 'href="/components"');
  assertStringIncludes(html, 'href="/docs/getting-started"');
  assertStringIncludes(html, 'href="https://github.com"');
});

Deno.test("HeroSection - includes background decorations", () => {
  const html = renderToString(HeroSection({}));

  // Check for background elements
  assertStringIncludes(html, "bg-gradient-to-br");
  assertStringIncludes(html, "bg-grid-pattern");
  assertStringIncludes(html, "blur-3xl");
});

Deno.test("HeroSection - includes SVG wave", () => {
  const html = renderToString(HeroSection({}));

  // Check for SVG wave element
  assertStringIncludes(html, "<svg");
  assertStringIncludes(html, 'viewBox="0 0 1440 120"');
});

Deno.test("HeroSection - has responsive design classes", () => {
  const html = renderToString(HeroSection({}));

  // Check for responsive classes
  assertStringIncludes(html, "lg:");
  assertStringIncludes(html, "xl:");
  assertStringIncludes(html, "sm:");
});

Deno.test("HeroSection - HTML snapshot", async (t) => {
  const html = renderToString(HeroSection({}));
  await assertSnapshot(t, html);
});
