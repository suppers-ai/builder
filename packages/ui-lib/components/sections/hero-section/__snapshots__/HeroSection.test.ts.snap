export const snapshot = {};

snapshot[`HeroSection - HTML snapshot 1`] = `'<section class="relative bg-gradient-to-br from-primary/10 via-secondary/5 to-accent/10 overflow-hidden"><div class="absolute inset-0 bg-grid-pattern opacity-5"></div><div class="absolute top-0 right-0 w-96 h-96 bg-primary/20 rounded-full blur-3xl -translate-y-48 translate-x-48"></div><div class="absolute bottom-0 left-0 w-96 h-96 bg-secondary/20 rounded-full blur-3xl translate-y-48 -translate-x-48"></div><div class="relative px-4 lg:px-6 py-20 lg:py-32"><div class="max-w-6xl mx-auto text-center"><div class="inline-flex items-center gap-2 bg-base-100 border border-base-300 rounded-full px-4 py-2 text-sm font-medium mb-8 shadow-sm"><div class="w-2 h-2 bg-success rounded-full animate-pulse"></div>New: Fresh 2.0 Support</div><h1 class="text-4xl lg:text-6xl xl:text-7xl font-bold text-base-content mb-6 leading-tight">Professional <span class="text-transparent bg-clip-text bg-gradient-to-r from-primary to-secondary">DaisyUI Components</span> for Fresh</h1><p class="text-lg lg:text-xl text-base-content/70 mb-8 max-w-3xl mx-auto leading-relaxed">The most comprehensive DaisyUI component library built specifically for Fresh 2.0. Get 65+ production-ready components with full TypeScript support, accessibility features, and seamless theming.</p><div class="flex flex-wrap justify-center gap-8 mb-12"><div class="text-center"><div class="text-2xl lg:text-3xl font-bold text-primary">65+</div><div class="text-sm text-base-content/60">Components</div></div><div class="text-center"><div class="text-2xl lg:text-3xl font-bold text-secondary">100%</div><div class="text-sm text-base-content/60">DaisyUI Coverage</div></div><div class="text-center"><div class="text-2xl lg:text-3xl font-bold text-accent">29</div><div class="text-sm text-base-content/60">Themes</div></div><div class="text-center"><div class="text-2xl lg:text-3xl font-bold text-success">TypeScript</div><div class="text-sm text-base-content/60">Full Support</div></div></div><div class="flex flex-col sm:flex-row gap-4 justify-center mb-12"><a href="/components" class="btn btn-primary btn-lg group">Browse Components<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-arrow-right"><path d="M5 12h14"></path><path d="m12 5 7 7-7 7"></path></svg></a><a href="/docs/getting-started" class="btn btn-outline btn-lg"><svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-code-xml"><path d="m18 16 4-4-4-4"></path><path d="m6 8-4 4 4 4"></path><path d="m14.5 4-5 16"></path></svg>Get Started</a><a href="https://github.com" class="btn btn-ghost btn-lg" target="_blank" rel="noopener noreferrer"><svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-github"><path d="M15 22v-4a4.8 4.8 0 0 0-1-3.5c3 0 6-2 6-5.5.08-1.25-.27-2.48-1-3.5.28-1.15.28-2.35 0-3.5 0 0-1 0-3 1.5-2.64-.5-5.36-.5-8 0C6 2 5 2 5 2c-.3 1.15-.3 2.35 0 3.5A5.403 5.403 0 0 0 4 9c0 3.5 3 5.5 6 5.5-.39.49-.68 1.05-.85 1.65-.17.6-.22 1.23-.15 1.85v4"></path><path d="M9 18c-4.51 2-5-2-7-2"></path></svg><span class="hidden sm:inline">View on GitHub</span><span class="sm:hidden">GitHub</span></a></div><div class="flex flex-wrap justify-center items-center gap-6 text-sm text-base-content/60"><div class="flex items-center gap-2"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-star"><polygon points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"></polygon></svg><span>Trusted by developers</span></div><div class="flex items-center gap-2"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-download"><path d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"></path><polyline points="7 10 12 15 17 10"></polyline><line x1="12" x2="12" y1="15" y2="3"></line></svg><span>Production ready</span></div><div class="flex items-center gap-2"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-zap"><path d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"></path></svg><span>Lightning fast</span></div><div class="flex items-center gap-2"><svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-palette"><circle cx="13.5" cy="6.5" r=".5" fill="currentColor"></circle><circle cx="17.5" cy="10.5" r=".5" fill="currentColor"></circle><circle cx="8.5" cy="7.5" r=".5" fill="currentColor"></circle><circle cx="6.5" cy="12.5" r=".5" fill="currentColor"></circle><path d="M12 2C6.5 2 2 6.5 2 12s4.5 10 10 10c.926 0 1.648-.746 1.648-1.688 0-.437-.18-.835-.437-1.125-.29-.289-.438-.652-.438-1.125a1.64 1.64 0 0 1 1.668-1.668h1.996c3.051 0 5.555-2.503 5.555-5.554C21.965 6.012 17.461 2 12 2z"></path></svg><span>Fully themeable</span></div></div></div></div><div class="absolute bottom-0 left-0 right-0"><svg viewBox="0 0 1440 120" fill="none" xmlns="http://www.w3.org/2000/svg" class="w-full h-auto"><path d="M0 120L60 105C120 90 240 60 360 45C480 30 600 30 720 37.5C840 45 960 60 1080 67.5C1200 75 1320 75 1380 75L1440 75V120H1380C1320 120 1200 120 1080 120C960 120 840 120 720 120C600 120 480 120 360 120C240 120 120 120 60 120H0Z" fill="currentColor" class="text-base-100"></path></svg></div></section>'`;
